### ISSUE TITLE
[Bug]: Removing "Open system Terminal in Current Folder" from Tools Settings

### ISSUE THREAD

**Issue by teropentis64** [mikekazakov/nimble-commander#378]
### Confirmation

- [X] I have read the contributing guidelines

### Version of Nimble Commander

Nimble 1.6.1 (4088), Sonoma 14.6.1

### Distribution Type

Downloaded directly from the website (including Brew)

### Expected Result

Removed desired tool from Tools list in settings panel.

### Actual Result

Removed desired tool from Tools list in settings panel AND removed different Tool shortcut icon in application toolbar at index 1.

### Steps to reproduce

1. Remove default "Open system Terminal in current folder" shortcut from application toolbar.
2. Add another one.
3. Go to Settings -> Tools and remove mentioned tool from the list.
4. Removing it from the Tools list will also remove shortcut at index 1 from application toolbar.

### Additional Information

In short: removing the default tool from Tools list removes the first tool shortcut in application toolbar (which doesn't correspond to the tool removed in Tools list).

P.S
Thank You for such amazing app. I've tried all of the File Managers and Nimble is really something else.

**Comment by mikekazakov** [comment]
Hi, thanks for reporting the issue, I've tried to reproduce and indeed something fishy is happening with the toolbar.

**Comment by mikekazakov** [comment]
The toolbar addresses the external tools using their index in the storage, which causes unexpected behaviour once tools are rearranged or some of them are removed. Now thinking about it, this approach seem to be rather silly. I'm migrating the external tools to be addressed via an internal UUID which will be stable, regardless of rearranging, renaming or removal.

**Comment by teropentis64** [comment]
Yea, I thought it's addressing them by index. I would like also to report that I found rather small misplace of button which I don't think really deserves additional issue report, but I bet You already noticed it.
Thanks again for Nimble!
![x](https://github.com/user-attachments/assets/90663c4e-7809-41f3-91e1-00be4793342f)

**Comment by mikekazakov** [comment]
Should be fixed now, please try the next nightly build.

**Comment by teropentis64** [comment]
I've ran Nightly Build #248, the toolbar items works as expected.

You actually noticed something that I didn't, and thats the button was a bit higher, good catch! This is indeed fixed too, but what I actually meant the first time is easier to see there:
![sc](https://github.com/user-attachments/assets/f160a480-b66e-4db3-b182-c557c517cdff)
When the button is selected with blue rectangle, it seem's like the downarrow button overlaps the Overwrite button. I think just adding few pixels padding on the left of downarrow button would do the job. Thanks!

**Comment by mikekazakov** [comment]
Ah, I see... I've shifted it by 1px so now they are just adjacent to each other, by tbh I'm not exactly sure what is considered "best practice" for such controls nowadays, i.e. should they be together or separated. Do you have any examples from other software?

**Comment by teropentis64** [comment]
I think nothing better fits macOS design than what Apple design in the first place 😄 I remember the down arrow button is visible on "Save As" window in various apps. I've measured it in 2 apps and seems to be the same. It's 14px (considering the grey 1px border outline is inside the button shape).
<img width="767" alt="border" src="https://github.com/user-attachments/assets/3bfef55b-2042-4a9f-a76a-21c0f24b9f93">

**Comment by mikekazakov** [comment]
Ok, done.

**Comment by teropentis64** [comment]
Thanks a lot!
